src/containers.tsx

```diff
@@ -11,12 +11,12 @@ export interface IDisabledContainerProps {
 }
 
 export interface IGuardedContainerProps {
-  disabledComponent: React.ComponentClass;
-  enabledComponent: React.ComponentClass;
+  disabledComponent: React.ReactNode;
+  enabledComponent: React.ReactNode;
   isGuarded: boolean;
 }
 
-export function createDisabledContainer (WrappedComponent: React.ComponentClass<any>): React.ComponentClass {
+export function createDisabledContainer (WrappedComponent: React.ComponentType<any>): React.ReactNode {
```

Comment:
`WrappedComponent: React.ReactNode` here wouldn't work, it would trigger issues w/ constructor call signature.